[5814] Create Group form allowing to create a group with empty space as name

jvega190 opened the pull request:
https://github.com/craftercms/craftercms/issues/5814

jvega190 committed:
Initial work - Create Group form allowing to create a group with empty space as name #5814

jvega190 committed:
fix translation variable name, add TODO for email input issue #5814

jvega190 committed:
Move isInvalidEmail and isInvalidUsername to UserManagement utils, fix validations issues #5814

jvega190 committed:
Add refs in CreateUserDialogContainer and move usernameRegex into util function #5814

rart commented on ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx:
```diff
@@ -116,6 +124,13 @@ const useStyles = makeStyles()((theme) => ({
+const translations = defineMessages({
+  invalidMinLength: {
+    id: 'createUserDialog.invalidMinLength',
+    defaultMessage: 'Min {length} characters.'
```
```suggestion
    defaultMessage: 'Min {length} characters'
```

rart commented on ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx:
```diff
@@ -148,7 +163,12 @@ export function CreateUserDialogContainer(props: CreateUserDialogContainerProps)
-        create(newUser)
+        const trimmedNewUser = {};
+        Object.entries(newUser).forEach(([key, value]) => {
+          trimmedNewUser[key] = typeof value === 'boolean' ? value : value.trim();
```
```suggestion
          trimmedNewUser[key] = typeof value === 'string' ? value.trim() : value;
```

rart commented on ui/app/src/components/EditUserDialog/utils.ts:
```diff
@@ -54,4 +55,7 @@ export interface EditUserDialogUIProps {
   onResetPassword(value: boolean): void;
+  isInvalidEmail(email: string): boolean;
+  validateRequiredField(field: string): boolean;
+  validateFieldMinLength(key: string, value: string): boolean;
```
Can we stop passing utils as props?

rart commented on ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx:
```diff
@@ -207,11 +240,19 @@ export function EditUserDialogUI(props: EditUserDialogUIProps) {
+                        validateRequiredField(user.email) ? (
+                          <FormattedMessage id="editUserDialog.emailRequired" defaultMessage="Email is required." />
+                        ) : isInvalidEmail(user.email) ? (
+                          <FormattedMessage id="editUserDialog.invalidEmail" defaultMessage="Email is invalid." />
```
```suggestion
                          <FormattedMessage id="editUserDialog.emailRequired" defaultMessage="Email is required" />
                        ) : isInvalidEmail(user.email) ? (
                          <FormattedMessage id="editUserDialog.invalidEmail" defaultMessage="Email is invalid" />
```

rart commented on ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx:
```diff
@@ -190,11 +212,22 @@ export function EditUserDialogUI(props: EditUserDialogUIProps) {
+                        validateRequiredField(user.lastName) ? (
+                          <FormattedMessage
+                            id="editUserDialog.lastNameRequired"
+                            defaultMessage="Last Name is required."
```
```suggestion
                            defaultMessage="Last Name is required"
```

rart commented on ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx:
```diff
@@ -173,11 +182,24 @@ export function EditUserDialogUI(props: EditUserDialogUIProps) {
+                        validateRequiredField(user.firstName) ? (
+                          <FormattedMessage
+                            id="editUserDialog.firstNameRequired"
+                            defaultMessage="First Name is required."
```
```suggestion
                            defaultMessage="First Name is required"
```

rart commented on ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx:
```diff
@@ -64,6 +65,10 @@ const translations = defineMessages({
+  },
+  invalidMinLength: {
+    id: 'userInfoDialog.invalidMinLength',
+    defaultMessage: 'Min {length} characters.'
```
```suggestion
    defaultMessage: 'Min {length} characters'
```

rart commented on ui/app/src/components/EditUserDialog/EditUserDialogContainer.tsx:
```diff
@@ -209,6 +235,9 @@ export function EditUserDialogContainer(props: EditUserDialogContainerProps) {
       onResetPassword={onResetPassword}
+      isInvalidEmail={isInvalidEmail}
+      validateRequiredField={validateRequiredField}
+      validateFieldMinLength={validateFieldMinLength}
```
avoid utils as props

rart commented on ui/app/src/components/EditGroupDialog/EditGroupDialogContainer.tsx:
```diff
@@ -212,12 +225,15 @@ export function EditGroupDialogContainer(props: EditGroupDialogContainerProps) {
       inProgressIds={inProgressIds}
       isDirty={isDirty}
+      validateRequiredField={validateRequiredField}
+      validateGroupNameMinLength={validateGroupNameMinLength}
```
Avoid utils as props

jvega190 committed:
Translations updates #5814

jvega190 committed:
PR review updates updates #5814

jvega190 committed:
Merge branch 'develop' into feature/5814

jvega190 committed:
Refactor MaxLength variables, add to EditUserDialogUI #5814

jvega190 commented:
@rart I moved the maxLength props for userDialogs to UserManagement/utils.ts for it to be consistent. I also added the maxLength restrictions to EditUserDialog (it was only added to CreateUserDialog)
